runtime/lua/vim/treesitter/query.lua

```diff
@@ -34,14 +72,45 @@ local function safe_read(filename, read_quantifier)
   return content
 end
 
+-- Disables loading query files for {lang} matching provided {patterns}.
+--
+-- If {patterns} is nil, it clears the ignore patterns.
+-- If {patterns} is string[], it replaces the currently set patterns (if any).
+--
+-- You can also pass a function as {patterns}, which will accept a table with
+-- currently set patterns as keys. You can modify that table and return a new
+-- table or nil. The return value will be used as the new ignore patterns.
+--
+---@param lang string
+---@param patterns nil | string[] | fun(patterns:table<string, boolean | nil>): nil | table<string, boolean | nil>
+function M.ignore_query_files(lang, patterns)
+  -- remove cached results
+  query_file_config.ignored_files[lang] = {}
+
+  if type(patterns) == "function" then
+    query_file_config.ignore_patterns[lang] = patterns(query_file_config.ignore_patterns[lang] or {})
+    return
+  end
+
+  if not patterns then
+    query_file_config.ignore_patterns[lang] = nil
+    return
+  end
+
+  query_file_config.ignore_patterns[lang] = {}
+  for _, pattern in ipairs(patterns) do
+    query_file_config.ignore_patterns[lang][pattern] = true
+  end
```

Comment:
Why not using a table directly here ? This is not the classic way of storing value here :wink:

Reply:
I did that to support this pattern:

```lua
vim.treesitter.ignore_query_files("lua", function(patterns)
  patterns["/nvim%-treesitter/"] = true -- or false/nil for removing
  return patterns
end)
```

I thought it might be handy when somebody wants to toggle a pattern on/off (via some keybinding or commands). Otherwise they have to loop over the table (`string[]`) and filter out the value... or check if the value already exists before adding the value.

Do you want store the table (as `string[]`) directy instead of making `table<string, boolean>`?

Reply:
To be honest yes, I'd rather keep it simple.
The simplest it is, the better.

Reply:
Updated 👍🏼